### PR TITLE
Ee 8289 simplify

### PR DESCRIPTION
**Pull request by Tim-Hammonds**

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceTest.java [hunk 1]
```diff
@@ -9,42 +9,47 @@
...
+                        entireNonAliasName,
+                        entireLastNameAndEachFirstName,
+                        multipleLastNames,
+                        namesWithFullStopSpaceCombinations,
+                        aliasCombinations,
+                        nameCombinations,
+                        specialCharacters
+                        ));
+
+        when(entireNonAliasName.generateCandidates(any(InputNames.class))).thenReturn(Collections.emptyList());
```
Comment (owainrobertsbjss) [inline src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceTest.java]: Could we perhaps loop over the list since they're all mocked equally?

File: src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java [hunk 2]
```diff
@@ -8,15 +9,17 @@
...
 @Component
+@Order(value = ALIAS_COMBINATIONS_STRATEGY_PRIORITY)
 public class AliasCombinations implements NameMatchingCandidateGenerator {
 
     @Override
     public List<CandidateName> generateCandidates(InputNames inputNames) {
         List<CandidateName> candidateNames = new ArrayList<>();
 
-        if (inputNames.aliasSurnames().isEmpty()) {
+        if (!inputNames.hasAliasSurnames()) {
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java]: I like the idea of calling all the generators, rather than having the service select them. But we need a unit test that AliasCombinations returns an empty list when there are no Aliases and, similarly NameCombinations returns an empty list when there are aliases.

File: src/main/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesService.java [hunk 3]
```diff
@@ -3,59 +3,33 @@
...
-
-        if (inputNames.hasAliasSurnames()) {
-            candidates.addAll(aliasCombinations.generateCandidates(inputNames));
-        } else {
-            candidates.addAll(nameCombinations.generateCandidates(inputNames));
-        }
+        InputNames inputNames = new InputNames(firstNames, lastNames, aliasSurnames);
 
-        candidates.addAll(specialCharacters.generateCandidates(inputNames));
+        List<CandidateName> candidates = candidateStrategies
```
Comment (owainrobertsbjss) [inline src/main/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesService.java]: We call the interface Generators not Strategies and we should be consistent. But I actually prefer Strategies as a name so maybe we can change it at some point?

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceIT.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
         SpecialCharacters.class,
         AliasCombinations.class,
         EntireNonAliasName.class,
+        EntireLastNameAndEachFirstName.class,
         NamesWithFullStopSpaceCombinations.class
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceIT.java]: Do we need a test for the ordering of the name combination generators?  Or is this implicitly tested in the BDD?

File: src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceTest.java (hunk 1 above)
Comment (Tim-Hammonds) [reply src/test/java/uk/gov/digital/ho/pttg/application/namematching/NameMatchingCandidatesServiceTest.java]: I did consider that, but I feel it is usually best to be explicit in how test collaborators are initialised.

File: src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java (hunk 2 above)
Comment (Tim-Hammonds) [reply src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java]: In the case of AliasCombinations, there was an existing test for this: shouldReturnEmptyListWhenNoAliasSurnames

Comment (Tim-Hammonds) [reply src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java]: In the case of NameCombinations, I had already added one: shouldReturnEmptyListWhenAliasSurnames
